### PR TITLE
Avoid sending "0 new commits" webhooks

### LINKED ISSUE #10498: Slack webhook showing message 0 new commits pushed
- Gitea version (or commit ref): 1.11.1
- Git version:
- Operating system:
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When deleting a branch after a PR merge, the following messages are printed from Slack webhook:

`...branchname] branch deleted by cornel`
`...branchname] 0 new commits pushed by cornel`

is the `0 new commits pushed` message intentional/needed?

### PR DESCRIPTION
**Pull request by S7evinK**
Avoids sending messages containing "0 new commits pushed" to services. Other solutions would probably break integration with eg. drone (see #11397)

Closes #10498

### CODE REVIEW

**Comment by lafriks** [comment]
wouldn't this result error being logged?

**Comment by S7evinK** [comment]
Yes, at the moment an error would be logged. Should this be a warning/info instead?

**Comment by lafriks** [comment]
Imho it should be handled somewhere in more central place and ignored silently

**Comment by S7evinK** [comment]
Done. Shouldn't result in anything being logged now.

File: modules/webhook/webhook.go [hunk 1]
```diff
@@ -76,6 +76,15 @@ func prepareWebhook(w *models.Webhook, repo *models.Repository, event models.Hoo
...
 	}
 
+	pushEvent, ok := p.(*api.PushPayload)
+	// Avoid sending "0 new commits" to non-integration relevant webhooks (e.g. slack, discord, etc.).
+	// Integration webhooks (e.g. drone) still receive the required data.
+	if ok && w.HookTaskType != models.GITEA && w.HookTaskType != models.GOGS {
+		if len(pushEvent.Commits) == 0 {
+			return nil
+		}
+	}
```
Comment (lafriks) [inline modules/webhook/webhook.go]:
```suggestion
	// Avoid sending "0 new commits" to non-integration relevant webhooks (e.g. slack, discord, etc.).
	// Integration webhooks (e.g. drone) still receive the required data.
	if pushEvent, ok := p.(*api.PushPayload); ok &&
		w.HookTaskType != models.GITEA && w.HookTaskType != models.GOGS &&
		len(pushEvent.Commits) == 0 {
		return nil
	}
```

**Review by lafriks — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
I hope this wont break things again - throug it looks promising

**Pull request merged**